packages/versioning/src/validate.ts

```diff
@@ -167,20 +169,96 @@ export function $onValidate(program: Program) {
 /**
  * Ensures that the version enum for a @versioned namespace has unique values.
  */
-function validateVersionEnumValuesUnique(program: Program, namespace: Namespace) {
-  const [_, versionMap] = getVersions(program, namespace);
-  if (versionMap === undefined) return;
+function validateVersionEnumDuplicates(
+  program: Program,
+  versionEnum: Enum,
+  versionMap: VersionMap
+) {
   const values = new Set(versionMap.getVersions().map((v) => v.value));
   if (versionMap.size !== values.size) {
-    const enumName = versionMap.getVersions()[0].enumMember.enum.name;
     reportDiagnostic(program, {
       code: "version-duplicate",
-      format: { name: enumName },
-      target: namespace,
+      format: { name: versionEnum.name },
+      target: versionEnum,
     });
   }
 }
 
+function splitSuffix(value: string): [string, string?] {
+  const betaSuffixes = ["-beta", "-preview"];
+  for (const suffix of betaSuffixes) {
+    if (value.endsWith(suffix)) {
+      return [value.slice(0, -suffix.length), suffix];
+    }
+  }
+  return [value, undefined];
+}
+
+/**
+ * Ensures that is a @versioned enum uses dates, the are in ascending order and
+ * that any preview of beta versions precede their non-beta counterparts.
+ */
+function validateVersionEnumDatesOrdering(program: Program, versionEnum: Enum) {
```

Comment:
I think this rule in this state is better to keep in Azure.Core. This is really targetting azure style of versioning

Reply:
None of the versioning decorators work correctly if the versions are not ordered, so why would this not be a core part of the versioning library?

Reply:
because the name of the version is up to the writer of the spec. It doesn't work if they mix their order sure but right now this is arbitrary to only enforce those dates on the generic versioning.